### ISSUE TITLE
Create a colo[u]r command

### ISSUE THREAD

**Issue by onerandomusername** [python-discord/sir-lancebot#677]
## Description

TL;DR add a command to get info on a color supplied by rgb, hex, possibly fuzzy matching by name, and possibly hsv or hsl. All of this would be local and no api usage whatsoever.
More details are below in the [`Proposed Implementation`](#Proposed-Implementation) section.

## Reasoning

Say one was in [`#media`](https://discord.com/channels/267624335836053506/753252897059373066/) and were trying to show a color for everyone, or were in offtopic and wanted to get color info for some graphic stuff or just wanted color info!
Relevant Convo in discord: [here](https://discord.com/channels/267624335836053506/635950537262759947/830930852493656094)

## Proposed Implementation

Some of the features would be:
- embedding an image of the color using [pillow](https://pypi.org/project/Pillow/) to create it
- converting the color to other formats using [colorsys](https://docs.python.org/3/library/colorsys.html#module-colorsys)
- giving the closest matching name from one of these two lists ([xkcd](https://xkcd.com/color/rgb/) and [this one](https://github.com/ryanzec/name-that-color/blob/master/lib/ntc.js#L116-L1681) suggested by joe), merged and stored locally.

Other features 

## Would you like to implement this yourself?

- [x] I'd like to implement this feature myself
- [x] Anyone can implement this feature

**Comment by jb3** [comment]
# Overview

Here are some features we should include:

- [ ] Colour information
- [ ] Random RGB or HSV colours
- [ ] Colour conversions
- [ ] Colour naming

These should *all* be implemented in the same command, but can be implemented separately, the above order is my recommendation for tackling them one by one.

# Details

## Colour information

Given a colour in one of the supported colour models (RGB/HSV/HSL/CMYK) we should return an embed with information on that colour.

Embed should include:
- The colour itself as both the embed colour and as the thumbnail.
- Colour conversions (see below).
- Colour name (see below).

### Example usage

```
.colour #6563C7
.colour cmyk(49, 50, 0, 22)
.colour hsl(241, 47, 58)
.colour rgb 101 99 199
```

## Random colours

Users should be able to fetch a random RGB colour or HSV colour. They should be able to do this for all supported colour models, which will probably be RGB/HSV/HSL/CMYK.

### Example usage

```
.colour random hsv
.colour random cmyk
.colour random rgb
```

## Colour conversions

Given an input in one of the supported colour models, we should convert to all other models. This information can be included in the [Colour information](#colour-information). 

Given the hex colour `#6563C7` we should convert to values such as:
- CMYK 49, 50, 0, 22
- RGB 101, 99, 199
- HSV 241, 50, 78
- HSL 241, 47, 58

We should most likely use fields for this, displaying the inputted colour format in the title and then a field for each conversion.

## Colour naming

We should identify the closest named colour to the inputted or generated colour, identifying things like `#6563C7` as approximately `Blue Violet Crayola` and so on.

We can use the aforementioned datasets to provide us with the data to identify close colours.

The implementation of this should take the inputted colour, convert it to RGB (which is what the above datasets store the names alongside) and then find the closest colour on all three RGB components.

**Comment by dawnofmidnight** [comment]
@onerandomusername Which parts of this are you interested in doing, since this is a rather large task.

**Comment by MrKomodoDragon** [comment]
@jb3 I would like to help

**Comment by ChrisLovering** [comment]
Hey @onerandomusername are you still looking to work on this? It seems like there are a bunch of people who would also like to help out.

**Comment by Xithrius** [comment]
Given that the author hasn't responded to the ping, this issue is now up for grabs.

**Comment by brad90four** [comment]
This seems like it is pretty well defined regarding the scope and the features required. I can try to give this a shot.

**Comment by CyberCitizen01** [comment]
> This seems like it is pretty well defined regarding the scope and the features required. I can try to give this a shot.

I would also like to work on this.

**Comment by Vthechamp22** [comment]
Another feature could be to generate color palettes?

**Comment by CyberCitizen01** [comment]
@Vthechamp22 Could you please elaborate a little more and If you can, please provide with some (visual) reference to what you are suggesting.
Thank you.
